### ISSUE TITLE
[HOLD for payment 2023-09-06] [$2000] Dev: Web - Console error on click of Next button in 2FA

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22929]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to Settings > Security > Two-factor authentication
2. Click on copy button and click on Next button
3. Scan the qr code in authenticator app
4. Enter six digit code
5. Click on Next button

## Expected Result:
It  should not show any console error

## Actual Result:
Console error appears (Failed prop type: Invalid prop `forwardedRef` supplied to `withOnyx(Component)`.)

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.40-5
**Reproducible in staging?:** n
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/9a290511-15ae-480b-bd0e-c428cae3da9c

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689328521600739

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d58d7cdae12240f0</li>
        <li>Upwork Job ID: 1684614195949875200</li>
        <li>Last Price Increase: 2023-08-08</li>
<li>Automatic offers: </li>
<ul>
<li>aimane-chnaif | Reviewer | 26032740</li>
<li>kushu7 | Contributor | 26032742</li>
<li>gadhiyamanan | Reporter | 26032744</li>
<li>tienifr | Contributor | 26054370</li>
</ul></li>
    </ul>
</details>

**Comment by muttmuure** [comment]
Closing dupe

**Comment by gadhiyamanan** [comment]
@muttmuure we can close the other issue and reopen this issue because this issue reported first in Slack

**Comment by gadhiyamanan** [comment]
@muttmuure https://github.com/Expensify/App/issues/22922 was closed, we can reopen this issue

**Comment by muttmuure** [comment]
Not overdue

**Comment by getusha** [comment]
Unable to reproduce on staging.

**Comment by hayata-suenaga** [comment]
@aimane-chnaif thank you for reviewing proposals 🙂

> I see there are several duplicated issues which have the same root cause and happen on different pages.
But this one is the first among open issues and also @kushu7 is the first to provide perfect solution.

Are these issues linked to this issue, and are people assigned to those duplicated issues aware that their issues will be fixed with this one?

**Comment by hayata-suenaga** [comment]
commented on #23201 thank you Aimane for the links and the investigation 🙇

**Comment by hayata-suenaga** [comment]
nice suggestion @aimane-chnaif 

We're actually migrating the entire application to TypeScript soon, so `forwardRef` will have type arguments in place in the same file as the component wrapped by `forwardRef`.

This is the example code: https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forward_and_create_ref/#:~:text=%3B%0A%20%20%7D%0A%7D-,forwardRef%3A,-import%20%7B%20forwardRef

**Comment by hayata-suenaga** [comment]
wait sorry I didn't read the context well.

why `withOnyx` is taking `forwardRef` instead of `ref`?

**Comment by hayata-suenaga** [comment]
ah found `forwardRef`: https://github.com/Expensify/react-native-onyx/blob/4ea9f77386bc2f6514694d04dd1f16278387b756/lib/withOnyx.js#L223-L227

**Comment by hayata-suenaga** [comment]
I believe we can remove type for `forwardedRef` [here](https://github.com/Expensify/react-native-onyx/blob/4ea9f77386bc2f6514694d04dd1f16278387b756/lib/withOnyx.js#L214-L217)

I think the right place to put the propType is the actual component returned from `forwardRef` [here](https://github.com/Expensify/react-native-onyx/blob/4ea9f77386bc2f6514694d04dd1f16278387b756/lib/withOnyx.js#L223)

But anyway, we're creating type declaration files in Onyx library and we can specify that the component returned from `withOnyx` takes a `ref` (if the wrapped component takes a `ref`). The returned component should actually take props that the wrapped component takes - props supplied by Onyx. 😓

**Comment by hayata-suenaga** [comment]
assigning @tienifr per this comment

**Comment by hannojg** [comment]
yeah for the latest only version to work you have to make this important change:

![Screenshot 2023-08-11 at 08 31 33](https://github.com/Expensify/App/assets/16821682/ba92c1dc-a0fe-48a4-afc5-b2df22838220)

Onyx will be updated to the latest version in this PR:

- https://github.com/Expensify/App/pull/24041/files#

(The PR that was linked earlier got reverted)

**Comment by muttmuure** [comment]
Not overdue

**Comment by muttmuure** [comment]
@kushu7 - $2000 - C
@tienifr - $2000 - C+
@gadhiyamanan - $250 - Reporter

**Comment by muttmuure** [comment]
Also paying @aimane-chnaif for the review

**Comment by muttmuure** [comment]
@aimane-chnaif made a good point that the PR was shared work between @kushu7 and @tienifr. 

The bounty should have been $3k between the two contributors, equalling $1.5k each. Since you've already been paid $2k for the work which is more than the bounty inclusive of urgency bonus, I will not be applying a further urgency bonus here. 

I have added an urgency bonus to @aimane-chnaif's upwork contract.
